src/components/SourceCode/CodeLine.tsx

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import styles from "./CodeLine.module.css";
+
+type CodeLineProps = {
+  line: string[];
+  withoutCoverage: boolean;
+  lineNumber: number;
+};
+
+const CodeLine: React.FC<CodeLineProps> = ({
+  line,
+  withoutCoverage,
+  lineNumber,
+}) => {
+  return (
+    <div className={styles.line}>
+      <span className={styles.lineNumber}>{lineNumber}</span>
+      <span
+        className={`${styles.code} ${
+          withoutCoverage ? styles.withoutCoverage : ""
+        }`}
+      >
+        {line.map((token, index) => (
+          <span key={index} className={styles.token}>
```

Comment:
Avoid using index as key when mapping an array, this key should be more unique value to avoid browser deletes some elements of other array.